packages/collaboration/src/sharedlink.ts

```diff
@@ -45,22 +45,458 @@ export async function showSharedLinkDialog({
+      );
+      this._users = this._users.concat(
+        groupsData
+          .filter((group: any) => !sharedGroupNames.has(group.name))
+          .map((group: any) => ({ name: group.name, type: 'group' }))
+      );
+    }
+
+    // Sort users and groups by name in alphabetical order
+    this._users.sort((a, b) => a.name.localeCompare(b.name));
+
+    this.updateSearchResults();
+  }
+
+  private async loadShares(): Promise<void> {
+    const sharesResponse = await fetch(
+      `${this._hubApiUrl}/shares/${this._serverOwner}/${this._serverName}`,
+      {
+        headers: {
+          Authorization: `token ${this._token}`
+        }
+      }
+    );
+    const sharesData = await sharesResponse.json();
+    this._shares = sharesData.items.map((item: any) => ({
+      name: item.user?.name || item.group?.name,
+      createdAt: item.created_at,
+      type: item.user ? 'user' : 'group'
+    }));
+  }
+
+  private async createShare(sharewith: any, type: any): Promise<void> {
+    // If the issuer can control the server, we add the "servers!server" scope to the share to let other users start/stop the server
+    const scopes = [
+      'access:servers!server=' + this._serverOwner + '/' + this._serverName
+    ];
+    if (this._canControlServer) {
+      scopes.push(
+        'servers!server=' + this._serverOwner + '/' + this._serverName
+      );
+    }
+
+    await fetch(
+      `${this._hubApiUrl}/shares/${this._serverOwner}/${this._serverName}`,
+      {
+        method: 'POST',
+        headers: {
+          Authorization: `token ${this._token}`,
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          [type]: sharewith.name,
+          scopes
+        })
+      }
+    );
+  }
+
+  private async deleteShare(sharewith: any, type: any): Promise<void> {
+    await fetch(
+      `${this._hubApiUrl}/shares/${this._serverOwner}/${this._serverName}`,
+      {
+        method: 'PATCH',
+        headers: {
+          Authorization: `token ${this._token}`
+        },
+        body: JSON.stringify({
+          [type]: sharewith.name
+        })
+      }
+    );
+  }
+
+  private populateBody(dialogBody: HTMLElement): void {
+    // Add search input
+    const searchContainer = document.createElement('div');
+    searchContainer.classList.add('search-container');
+    this._searchInput = document.createElement('input');
+    this._searchInput.type = 'text';
+    this._searchInput.classList.add('search-input');
+    this._searchInput.placeholder = this._trans.__(
+      'Type to search for a user or a group to share your server with...'
+    );
+    this._searchInput.addEventListener('input', () => {
+      this.updateSearchResults();
+    });
+    searchContainer.appendChild(this._searchInput);
+    dialogBody.appendChild(searchContainer);
+
+    // Add search results container
+    this._searchResults = document.createElement('div');
+    this._searchResults.classList.add('search-results');
```

Comment:
Let's prefix the class names to avoid conflicts with other extensions and user code:

```suggestion
    this._searchResults.classList.add('jp-ManageSharesBody-search-results');
```